graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/kotlin/KotlinDataTypeGenerator.kt

```diff
@@ -53,23 +53,21 @@ class KotlinInputTypeGenerator(private val config: CodeGenConfig, private val do
     fun generate(definition: InputObjectTypeDefinition, extensions: List<InputObjectTypeExtensionDefinition>): KotlinCodeGenResult {
 
         val fields = definition.inputValueDefinitions
-            .filter(ReservedKeywordFilter.filterInvalidNames)
-            .map {
-            val defaultValue: Any
-            if (it.defaultValue != null) {
-                defaultValue = when (it.defaultValue) {
-                    is BooleanValue -> (it.defaultValue as BooleanValue).isValue
-                    is IntValue -> (it.defaultValue as IntValue).value
-                    is StringValue -> (it.defaultValue as StringValue).value
-                    is FloatValue -> (it.defaultValue as FloatValue).value
-                    else -> it.defaultValue
-                }
-
-                Field(it.name, typeUtils.findReturnType(it.type), typeUtils.isNullable(it.type), defaultValue)
-            } else {
-                Field(it.name, typeUtils.findReturnType(it.type), typeUtils.isNullable(it.type))
-            }
-        }.plus(extensions.flatMap { it.inputValueDefinitions }.map { Field(it.name, typeUtils.findReturnType(it.type), typeUtils.isNullable(it.type)) })
+                .filter(ReservedKeywordFilter.filterInvalidNames)
+                .map {
+                    val type = typeUtils.findReturnType(it.type)
+                    val defaultValue = it.defaultValue?.let { defVal ->
+                        when (defVal) {
+                            is BooleanValue -> CodeBlock.of("%L", defVal.isValue)
+                            is IntValue -> CodeBlock.of("%L", defVal.value)
+                            is StringValue -> CodeBlock.of("%S", defVal.value)
+                            is FloatValue -> CodeBlock.of("%L", defVal.value)
```

Comment:
I think this needs to be `CodeBlock.of("%Lf")` to generate a valid float literal
e.g.:

```kotlin

CodeBlock.of("%L", 0.1f) // produces the literal 0.1, a Double in Kotlin
```

Reply:
Fixed

Reply:
After giving it some more thought, I think that the initial version of using double is correct as the [spec](http://spec.graphql.org/June2018/#sec-Float) calls them floats, but use doubles. [KotlinTypeUtils](https://github.com/Netflix/dgs-codegen/blob/9ff5d977d2fdb68d12a84b7e0e007c8a8bd1cf9b/graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/kotlin/KotlinTypeUtils.kt#L105) also maps the GraphQL type `Float` to Kotlin's `Double`.

What do you think?

Reply:
Yes, internally we map `Float` to Kotlin's `Double`. So I think it makes sense to have it just as `CodeBlock.of("%Lf")`

Reply:
I've removed the `f` again (though your last two comments are contradicting).

Reply:
Sorry for the confusion. I did mean CodeBlock.of("%L")